transport/src/main/java/io/netty5/channel/ChannelHandler.java

```diff
@@ -322,6 +330,18 @@ default Future<Void> close(ChannelHandlerContext ctx) {
         return ctx.close();
     }
 
+    /**
+     * Called once a shutdown operation is made.
+     *
+     * @param ctx               the {@link ChannelHandlerContext} for which the shutdown operation is made
+     * @param direction         the {@link ChannelShutdownDirection} that is used.
+     * @return                  the {@link Future} which will be notified once the operation completes.
+     */
+    @Skip
+    default Future<Void> shutdown(ChannelHandlerContext ctx, ChannelShutdownDirection direction) {
```

Comment:
The javadocs could be more clear about when either method is called. From the text and types they look very similar.

Reply:
any suggestion ?

Reply:
In this one we can say that "we" are requesting a shutdown, and reference the outbound invoker method.
And in the other we can say that the peer requested a shutdown.
Does that make sense?